rcl/src/rcl/logging.c

```diff
@@ -107,6 +113,16 @@ rcl_logging_configure(const rcl_arguments_t * global_args, const rcl_allocator_t
 rcl_ret_t rcl_logging_fini()
 {
   rcl_ret_t status = RCL_RET_OK;
+
+  if (rcutils_atomic_load_uint64_t(&__rcl_logging_init_count) > 0) {
+    uint64_t current_count = rcutils_atomic_fetch_add_uint64_t(&__rcl_logging_init_count, -1);
```

Comment:
Assuming we will stick with reference counting...

The condition above becomes true if this line is executed when `__rcl_logging_init_count` is zero, which can happen since the above condition and this fetch/add are not an atomic action together. You should probably not use an unsigned int, and you should probably not have a two step check, instead preferring to always subtract and add one later if it ended up being less than zero.

Reply:
This comment is not needed anymore since we're not concerned with thread-safety (I thought that was what you guys were doing here), but I think you don't need the atomic in that case.